src/types.ts

```diff
@@ -0,0 +1,165 @@
+ * @see https://docs.givechariot.com/integrating-connect#pre-populate-data-into-your-connect-session
+ */
+export type DonationRequestParams = Partial<{
+	/**
+	 * The donation amount in cents. e.g., for a $20 donation, enter 2000.
+	 */
+	amount: number
+	/**
+	 * The donor’s first name
+	 */
+	firstName: string
+	/**
+	 * The donor’s last name
+	 */
+	lastName: string
+	/**
+	 * The donor’s email address. Must be in email format
+	 */
+	email: string
+	/**
+	 * The donor’s phone number. Please provide the phone number with the country code and without special characters
+	 */
+	phone: string
+	/**
+	 * A note the donor wants to send to the nonprofit
+	 */
+	note: string
+	/**
+	 * Indicates if this donation should be sent anonymously (default: false)
+	 */
+	anonymous: boolean
+	/**
+	 * The designation to include on the grant. If this is left blank, “Where needed most” will be used. Note that including a custom designation may cause the grant approval process to take longer. Designations over 100 characters will be truncated.
+	 */
+	designation: string
+	/**
+	 * The donor’s address: line1, line2, city, state, postalCode
+	 */
+	address: {
+		line1: string
+		line2: string
+		city: string
+		state: string
+		postalCode: string
+	}
+	/**
+	 * An object with a set of name-value pairs. You can use this object to include any miscellaneous information you want to tie to the workflow session.
+	 */
+	metadata: Record<string, unknown>
+	/**
+	 * The ID of the preselected DAF provider.
+	 */
+	fundId: string
+	/**
+	 * The recurring frequency of the DAF grant (if it’s a recurring gift). This parameter allows the following possible enum values: ONE_TIME, MONTHLY. If not provided, the grant will default to a ONE_TIME grant. For more information on recurring donations, please see the section below
+	 */
+	frequency: "ONE_TIME" | "MONTHLY"
+}>
+
+export type DonationRequestReturnType =
+	| DonationRequestParams
+	| false
+	// biome-ignore lint/suspicious/noConfusingVoidType: This is a valid use case for void
+	| void
+	| Promise<DonationRequestReturnType>
+
+/**
+ * @see https://docs.givechariot.com/button-styles#creating-a-custom-theme
+ */
+export type CustomTheme = {
+	/**
+	 *  The width of the button up to `16` or `64px`
+	 */
+	width: number
+	/**
+	 * 	The height of the button up to `16` or `64px`
+	 */
+	height: number
+	/**
+	 * 	Show an extended text (Give with Donor Advised Fund) version of the DAFpay button text
+	 */
+	showExtendedText: boolean
+}
+
+/**
+ * @see https://docs.givechariot.com/button-styles#using-a-default-theme
+ */
+export type ChariotTheme =
+	| "DefaultTheme"
+	| "LightModeTheme"
+	| "LightBlueTheme"
+	| "GradientTheme"
+
+export type SuccessEvent = CustomEvent<{
+	workflowSessionId: string
+	grant: {
```

Comment:
I think this needs to be updated. We can either return a `grant`, `grantIntent`, `recurringGrant` or `recurringGrantIntent`. see: https://docs.givechariot.com/integrating-connect#recurring-donations-beta

Reply:
I'm seeing the following for the intents. Is this all that's provided? Asking because wasn't able to find a schema for the `grant` field, just copied from the console.

```
  {
    "workflowSessionId":"79e772be-547d-4c9c-8b76-4ac4ed4c441a", //Id of the Connect session
    "recurringGrantIntent": {
      "userFriendlyId": "100020", //The user-friendly identifier for the grant intent
      "fundId": "bbf485dd-a056-4a9d-89a8-06e201cdbf7f", //Id of the donor advised fund
      "amount": 2000, //The grant amount expressed in units of cents; USD only
      "frequency": "MONTHLY", // The frequency of the recurring schedule for the donation
      "metadata": {} // The same metadata object that was passed in the onDonationRequest callback referenced above
    }
  }

```

and 

```
{
    "workflowSessionId":"79e772be-547d-4c9c-8b76-4ac4ed4c441a", //Id of the Connect session
    "grantIntent": {
      "userFriendlyId": "100020", //The user-friendly identifier for the grant intent
      "fundId": "bbf485dd-a056-4a9d-89a8-06e201cdbf7f", //Id of the donor advised fund
      "amount": 2000, //The grant amount expressed in units of cents; USD only
      "metadata": {} // The same metadata object that was passed in the onDonationRequest callback referenced above
    }
  }
```

Also, assuming `recurringGrant` is the same schema as `grant`?